lstchain/reco/r0_to_dl1.py

```diff
@@ -673,3 +650,41 @@ def rescale_dl1_charge(event, scaling_factor):
 
     for tel_id, tel in event.dl1.tel.items():
         tel.image *= scaling_factor
+
+def interpolate_dl1a_of_bad_pixels(event, subarray, telescope_id):
+
+    """
+    interpolate dl1a parameters (charge and pulse peak time) of bad pixels using neighboring pixels
+
+    Parameters
+    ----------
+    
+    event: `ctapipe.containers.ArrayEventContainer`
+    subarray: `ctapipe.instrument.subarray.SubarrayDescription`
+    telescope_id: `int`
+    """
+    selected_gain = event.r1.tel[telescope_id].selected_gain_channel
+    bad_pixels = event.mon.tel[telescope_id].calibration.unusable_pixels[selected_gain] 
```

Comment:
This looks wrong. You seem to select the gain twice here.

Reply:
yeah... you are right... 
Removed the duplication.